coredns/gateway/controller.go

```diff
@@ -263,17 +264,36 @@ func (c *Controller) getClusterStatusMap() map[string]bool {
 	return c.clusterStatusMap.Load().(map[string]bool)
 }
 
-func (c *Controller) getCheckedClientset(kubeConfig *rest.Config) (dynamic.ResourceInterface, error) {
+func (c *Controller) getCheckedClientset(kubeConfig *rest.Config) (dynamic.ResourceInterface, bool, error) {
```

Comment:
We’ve said previously that we want to avoid adding boolean return values to represent information which is available elsewhere; in this instance, returning `false` implies returning a `nil` clientset.

Reply:
But that means returning `nil, nil` in some cases which is not idiomatic, ie typically a nil error means the first return value is valid. Plus we have he `nilnill` linter which is intended to flag that but, after changing the code, it didn't. The other solution is to return an error indicating "not found". I'll do that.